Sources/Spreadbot.swift

```
@@ -0,0 +1,410 @@
+            
+            if let sseConnection = strongSelf.sseConnection {
+                sseConnection.close()
+                strongSelf.print("SpreadbotLog: SSE connection closed")
+                strongSelf.sseConnection = nil
+            }
+            
+            strongSelf.print("SpreadbotLog: Attempting to poll for data..")
+            
+            // Create a scheduled timer
+            strongSelf.pollingTimer = Timer(
+                timeInterval: 15,
+                target: strongSelf,
+                selector: #selector(strongSelf.pollReq),
+                userInfo: nil,
+                repeats: true
+            )
+            // Add the timer to a runloop (in this case the main run loop)
+            RunLoop.main.add(strongSelf.pollingTimer!, forMode: RunLoopMode.commonModes)
+        }
+    }
+    
+    private func sseDidReceiveMessage() {
+        self.sseConnection?.onMessage { [weak self] (id, event, data) in
+            guard let strongSelf = self else { return }
+            
+            DispatchQueue.main.async { _ in    
+                strongSelf.print("SpreadbotLog: SSE message received)")
+                
+                guard id != nil && event != nil && data != nil else { return }
+                
+                strongSelf.lastIteration = id!
+                strongSelf.onMessage?(strongSelf.lastIteration, event!, data!)
+            }
+        }
+    }
+    
+    // MARK: - HTTP Polling Request
+    
+    @objc private func pollReq() {
+        getData(
+            onSuccess: { [weak self] data in
+                guard let strongSelf = self else { return }
+
+                strongSelf.onOpen?()
+                strongSelf.delegate?.spreadbotDelegateDidConnect()
+                
+                guard let message = data as? (id: String, event: String, data: Data) else { return }
+                guard Int(message.id)! >= Int(strongSelf.lastIteration)! else { return }
```

I understand that for technical reasons, you are transporting an Int id as a String. This is fine, but I recommend to get rid of the String ASAP, i.e. have it fail early. Ideally, your parse function would already spit out an `id: Int` (or fail with "Unparsable message" error). This is the strict Swift approach which takes some getting used to, but will eliminate some potential bugs.